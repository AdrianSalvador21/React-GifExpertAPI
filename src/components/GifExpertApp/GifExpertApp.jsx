import React, {useState} from 'react';
import AddCategory from "../AddCategory/AddCategory";
import GifGrid from "../GifGrid/GifGrid";

const GifExpertApp = ({defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <h2>GifExpertApp</h2>

      <AddCategory setCategories={setCategories}/>
      <hr />
      <ol>
        {
          categories.map((category, idx) =>
            (
              <GifGrid category={category}
                       key={category}
              />
            )
          )
        }
      </ol>
    </div>
  );
};

export default GifExpertApp;
