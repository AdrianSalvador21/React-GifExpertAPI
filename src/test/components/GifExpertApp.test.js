import '@testing-library/react';
import '@testing-library/jest-dom';
import {shallow} from "enzyme";
import React from "react";
import GifExpertApp from "../../components/GifExpertApp/GifExpertApp";

jest.mock('../../hooks/useFetchGifs');

describe('TEST GifExpertApp', () => {
  const wrapper = shallow(<GifExpertApp />);
  test('should show component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show list of categories', () => {
    const categories = ['Goku', 'Vegetta'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  });


});
