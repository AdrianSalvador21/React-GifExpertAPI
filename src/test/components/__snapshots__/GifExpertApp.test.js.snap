// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST GifExpertApp should show component correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`TEST GifExpertApp should show list of categories 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Goku"
      key="Goku"
    />
    <GifGrid
      category="Vegetta"
      key="Vegetta"
    />
  </ol>
</div>
`;
